Makefile

```diff
@@ -0,0 +1,53 @@
+# Settings
+# --------
+
+export BUILD_DIR:=$(CURDIR)/.build
+
+K_SUBMODULE:=$(BUILD_DIR)/k
+
+.PHONY: all clean deps k-deps distclean clean-submodules kore-exec
+
+all: kore-exec
+
+test: kore-tests k-tests
+
+kore-tests:
+	stack test --pedantic
+
+k-tests: kore-exec k-deps
+	$(MAKE) -C src/main/k/working test
+
+kore-exec: $(BUILD_DIR)/bin/kore-exec
+
+$(BUILD_DIR)/bin/kore-exec:
+	stack install kore:exe:kore-exec \
+		--local-bin-path $(abspath $(BUILD_DIR))/bin
+
+deps: k-deps
+
+
+distclean: clean
+	cd $(K_SUBMODULE) \
+		&& mvn clean -q
+	git submodule deinit --force -- ./
+
+clean: clean-submodules
+	stack clean
+	$(MAKE) -C src/main/k/working clean
+	rm -rf $(BUILD_DIR)/bin
+
+clean-submodules:
+	rm -rf $(K_SUBMODULE)/make.timestamp
+
+
+k-deps: $(K_SUBMODULE)/make.timestamp
+
+$(K_SUBMODULE)/make.timestamp:
```

Comment:
What does this timestamp do?

Reply:
it marks that K was built.  I was inspired from the identical mechanism used by the `evm-semantics`